django/contrib/auth/middleware.py

```diff
@@ -165,9 +170,26 @@ def __call__(self, request):
             # by logging the user in.
             request.user = user
             auth.login(request, user)
-        return self.get_response(request)
 
     async def __acall__(self, request):
+        # RemovedInDjango61Warning.
+        if (
+            self.__class__.process_request is not RemoteUserMiddleware.process_request
+            and self.__class__.aprocess_request is RemoteUserMiddleware.aprocess_request
+        ):
```

Comment:
If this guard is changed so it always evaluates to `True`, all tests still pass. This means that we may be missing a test to ensure that the warning is only emitted in the proper context.